### PR TITLE
Make `--help` and `--version` work with `allowUnknownFlags: false`

### LINKED ISSUE #197: `--help` and `--version` should work with `allowUnknownFlags: false`
As long as `{autoHelp: true}` or `{autoVersion: true}`.

Currently, it fails:

```
❯ node cli.js --help
Unknown flag
--help
```

---

// @weareoutman

### PR DESCRIPTION
**Pull request by msabramo**
Make `--help` and `--version` work with `allowUnknownFlags: false`

Fixes #197

Issue originally discovered with `chalk-cli` in https://github.com/chalk/chalk-cli/issues/32

### CODE REVIEW

**Comment by sindresorhus** [comment]
Can you add tests? 🙏

**Comment by msabramo** [comment]
Yeah I thought of that right after I submitted this. 😄

**Comment by msabramo** [comment]
See 368e760 for tests.

Unfortunately, I think my change broke 20 other unrelated existing tests, so I may have to think of a better way to accomplish the fix without affecting other stuff.

**Comment by msabramo** [comment]
OK, I got all the tests passing, but honestly I feel like I was hacking around without really understanding what the behavior should  be. So someone should take a close look and make sure I didn't mess things up.

File: index.js [hunk 1]
```diff
@@ -129,6 +129,19 @@ const meow = (helpText, options = {}) => {
 		...options,
 	};
 
+	let addedAutoHelp = false;
+	let addedAutoVersion = false;
+
+	if (options.autoHelp && !options.flags.help) {
+		options.flags.help = {type: 'boolean'};
```
Comment (sindresorhus) [inline index.js:136]: You are modifying the user's input object here. You should not do that.

File: index.js [hunk 2]
```diff
@@ -129,6 +129,19 @@ const meow = (helpText, options = {}) => {
 		...options,
 	};
 
+	let addedAutoHelp = false;
```
Comment (sindresorhus) [inline index.js:132]:
```suggestion
	let didAddAutoHelp = false;
```

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
> So someone should take a close look and make sure I didn't mess things up.

That someone is me and I don't think I'll time to review this in detail for quite some time, unfortunately.

File: test/fixtures/fixture-allow-unknown-flags.js [hunk 3]
```diff
@@ -16,4 +16,12 @@ const cli = meow({
 	},
 });
 
+export const expectedHelp = `
+  Custom description
+
+  Usage
+    foo <input>
+
```
Comment (LitoMore) [inline test/fixtures/fixture-allow-unknown-flags.js:24]:
```suggestion
```

**Review by LitoMore — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
Closing for lack of activity.

**Pull request closed without merging**